### PR TITLE
MC Runlevels

### PR DESCRIPTION
**Pull request by Cyberboss**
Instead of splitting subsystems by "fire in lobby" and "fire normally" we can now set a range for subsystems to fire in: LOBBY, SETUP, GAME, POSTGAME. Ticker subsystems still always run. This makes things more general for StonedMC apps that don't have a clear cut place to define lobby vs round.

@MrStonedOne

### CODE REVIEW

**Comment by MrStonedOne** [comment]
run levels should be flags so you can specify something that say, only runs at init or only runs in post game, or something that doesn't run in post game but all others.

**Comment by MrStonedOne** [comment]
>StonedMC apps

You fucking wot?

You call SS's 'apps' again and i'll hunt you down i swear on me mom

**Comment by Fox-McCloud** [comment]
>You call SS's 'apps' again and i'll hunt you down i swear on me mom

What about widgets?

**Comment by Cyberboss** [comment]
I mean BYOND apps

**Comment by MrStonedOne** [comment]
Still, the use of the word apps violates the geneva convention section on torture.

**Comment by Fox-McCloud** [comment]
😛

**Comment by Iamgoofball** [comment]
Help the shuttle app keeps refusing to launch custom bought shuttles

File: code/__DEFINES/MC.dm [hunk 1]
```diff
@@ -20,8 +20,9 @@
 
 //SubSystem flags (Please design any new flags so that the default is off, to make adding flags to subsystems easier)
 
+//deprecated, can be replaced
 //subsystem should fire during pre-game lobby.
-#define SS_FIRE_IN_LOBBY 1
+//#define SS_FIRE_IN_LOBBY 1
```
Comment (lzimann) [inline code/__DEFINES/MC.dm]: why leaving this commented out

Comment (MrStonedOne) [inline code/__DEFINES/MC.dm]: I mean, I think to show that the flag is open, but he could just re-shift them around, the point of using defines is you can do that without side effects

**Pull request merged**